paddle/fluid/operators/set_value_op.h

```diff
@@ -248,7 +251,7 @@ class SetValueKernel : public framework::OpKernel<T> {
     if (value_tensor != nullptr) {
       // ElementwiseComputeEx can do broadcasting
       ElementwiseComputeEx<SubFunctor<T>, DeviceContext, T>(
-          ctx, &slice_t, value_tensor, -1, SubFunctor<T>(), &slice_t);
+          ctx, &zeros_t, value_tensor, -1, SubFunctor<T>(), &slice_t);
```

Comment:
The shape of slice_t and value_tensor may not equal, which result in the error, please check it.

Reply:
Yes, but `ElementwiseComputeEx` support broadcast and the check will be done when call `ElementwiseComputeEx`.

Reply:
If slice_t.shape=[3,1] and zeros.shape=[3,1] and value.shape=[3], then the out tensor of ElementwiseComputeEx should be of shape [3,3]. If slice_t not re-allocate memory, it may result in error.

Reply:
Thanks, done